narwhals/dtypes.py

```diff
@@ -698,4 +698,23 @@ def __repr__(self) -> str:
 
 
 class Date(TemporalType):
-    """Data type representing a calendar date."""
+    """Data type representing a calendar date.
+
+    Examples:
+       >>> import pandas as pd
+       >>> import polars as pl
+       >>> import pyarrow as pa
+       >>> import narwhals as nw
+       >>> from datetime import date, timedelta
+       >>> data = [date(2024, 12, 1) + timedelta(days=1) for d in range(1, 4)]
```

Comment:
I have a tiny nitpick:

```suggestion
       >>> data = [date(2024, 12, 1) + timedelta(days=d) for d in range(4)]
```

Reply:
Thank you! I overlooked this!